lib/usecases/run/GetAllRunsUseCase.js

```diff
@@ -78,16 +78,33 @@ class GetAllRunsUseCase {
 
             if (runNumbers) {
                 const runNumberList = runNumbers.split(SEARCH_ITEMS_SEPARATOR)
-                    .map((runNumber) => parseInt(runNumber, 10))
-                    .filter((runNumber) => !Number.isNaN(runNumber));
+                    .map((runNumber) => runNumber.trim())
+                    .filter(Boolean);
 
-                if (runNumberList.length) {
-                    if (runNumberList.length > 1) {
-                        // Multiple run numbers.
-                        filteringQueryBuilder.where('runNumber').oneOf(...runNumberList);
+                const allRuns = new Set();
+
+                runNumberList.forEach((runNumber) => {
+                    if (runNumber.includes('-')) {
+                        const [start, end] = runNumber.split('-').map((n) => parseInt(n, 10));
+                        if (!Number.isNaN(start) && !Number.isNaN(end) && start <= end) {
```

Comment:
The last test is actually already covered by the loop

Reply:
What do you mean by last test?

Reply:
I mean `start <= end` is included in the loop. You will not enter the loop if start > end

Reply:
done